tests/contrib/botocore/test.py

```diff
@@ -3765,3 +3774,414 @@ def test_schematized_unspecified_service_secretsmanager_v1(self):
+
+    @pytest.mark.snapshot(ignores=snapshot_ignores)
+    @mock_s3
+    def test_aws_payload_tagging_s3_invalid_config(self):
+        with self.override_config(
+            "botocore",
+            dict(payload_tagging_request="non_json_path", payload_tagging_response="$..Attr ibutes.PlatformCredential"),
+        ):
+            s3 = self.session.create_client("s3", region_name="us-west-2")
+            Pin(service=self.TEST_SERVICE, tracer=self.tracer).onto(s3)
+
+            s3.list_buckets()
+            s3.list_buckets()
+
+            spans = self.get_spans()
+            assert spans
+            span = spans[0]
+            assert len(spans) == 2
+            assert_is_measured(span)
+            assert span.get_tag("aws.operation") == "ListBuckets"
+            assert span.get_tag("component") == "botocore"
+            assert span.get_tag("span.kind"), "client"
+            assert_span_http_status_code(span, 200)
+            assert span.service == "test-botocore-tracing.s3"
+            assert span.resource == "s3.listbuckets"
+
+            assert not span._links, "no links, i.e. no span pointers"
+
+            # testing for span error
+            self.reset()
+            try:
+                s3.list_objects(bucket="mybucket")
+            except Exception:
+                spans = self.get_spans()
+                assert spans
+                span = spans[0]
+                assert span.error == 1
+                assert span.resource == "s3.listobjects"
+
+    @pytest.mark.snapshot(ignores=snapshot_ignores)
+    @mock_s3
+    def test_aws_payload_tagging_s3_valid_config(self):
+        with self.override_config(
+            "botocore", dict(payload_tagging_request="$..bucket", payload_tagging_response="$..HTTPHeaders")
+        ):
+            s3 = self.session.create_client("s3", region_name="us-west-2")
+            Pin(service=self.TEST_SERVICE, tracer=self.tracer).onto(s3)
+
+            s3.list_buckets()
+            s3.list_buckets()
+
+            spans = self.get_spans()
+            assert spans
+            span = spans[0]
+            assert len(spans) == 2
+            assert_is_measured(span)
+            assert span.get_tag("aws.operation") == "ListBuckets"
+            assert span.get_tag("component") == "botocore"
+            assert span.get_tag("span.kind"), "client"
+            assert_span_http_status_code(span, 200)
+            assert span.service == "test-botocore-tracing.s3"
+            assert span.resource == "s3.listbuckets"
+
+            assert not span._links, "no links, i.e. no span pointers"
+
+            # testing for span error
+            self.reset()
+            try:
+                s3.list_objects(bucket="mybucket")
+            except Exception:
+                spans = self.get_spans()
+                assert spans
+                span = spans[0]
+                assert span.error == 1
+                assert span.resource == "s3.listobjects"
```

Comment:
I'm assuming we always expect an Exception to be raised here (because the bucket doesn't exist)? If so, could we do:
```suggestion
            with pytest.raises(Exception):
                s3.list_objects(bucket="mybucket")
            spans = self.get_spans()
            assert spans
            span = spans[0]
            assert span.error == 1
            assert span.resource == "s3.listobjects"
```
If there's a specific type of error we should look for, that could replace the general `Exception` in the context manager.

Reply:
Thanks, swapped to

```
with pytest.raises(Exception):
                s3.list_objects(bucket="mybucket")
```

and removed the asserts as it is all snapshotted